azurelinuxagent/common/logcollector.py

```diff
@@ -0,0 +1,377 @@
+# Microsoft Azure Linux Agent
+#
+# Copyright 2020 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+
+import glob
+from heapq import heappush, heappop
+import logging
+import os
+import subprocess
+import time
+import zipfile
+
+# Please note: be careful when adding agent dependencies in this module.
+# This module uses its own logger and logs to its own file, not to the agent log.
+
+from azurelinuxagent.common.conf import get_lib_dir, get_ext_log_dir
+from azurelinuxagent.common.future import ustr
+
+_EXTENSION_LOG_DIR = get_ext_log_dir()
+_AGENT_LIB_DIR = get_lib_dir()
+_AGENT_LOG = "/var/log/waagent.log"
```

Comment:
Similar to `get_lib_dir()` function, could you also add a function `get_agent_log_file()` or something and replace all occurances of `/var/log/waagent.log` with it? 
Or should that be done in a separate PR?

Reply:
+1

Reply:
I initially wanted to tackle it in a separate PR, but might as well.